### PR TITLE
Update product SEO metadata length

### LINKED ISSUE #34925: New Product Page vs. Old product page; maxchar for meta title and description
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

_Old product page:_ maximum characters for meta title and meta description is 70 and 160 chars. Although **more characters can be saved** when old product page is used. The table accepts 128 characters for meta title and 512 characters for meta description. 

_New Product Page:_ maximum characters for meta title and meta description is 70 and 160 chars. **Saving page with more characters can't not be done** and **information is showing under the form informing than only 70/160 characters can be used**.

The different behaviours of those two pages, old vs. new, is complicating updating products in the future. When using old page, longer titles and longer desctiptions can be saved. Those longer titles and desctiptions can be showed in the New Product Page, but then the product can't be updated with other info, unless the title and description has been decreased to 70/160 characters.

For some older shop owners this will complictate things, they will have to update all meta titles and description in order to use the New Product Page, just to be able to update simple things like stock amount.

This change in the New PRoduct Page will be a massive problems for many updated shop and owners.

Should it be better if the **product can be saved anyway**, but **with warning text that recommendation** is that meta title should be maximum 70 chars and meta description 160 chars? 

### Expected behavior
x the number of characters in the field.

- for the meta_title

Display on the bottom of the field : "x of 70 characters used (recommended)"
Validation server: 255 characters, like on the previous page.

- for the meta_description

Display on the bottom of the field : "x of 160 characters used (recommended)"
Validation server: 512 characters, like on the previous page.

### Steps to reproduce

Swithing between old and new product page, shows clearly what can be saved.

### PrestaShop version(s) where the bug happened

8.1.0, 8.1.2

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Update product SEO metadata length
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See https://github.com/PrestaShop/PrestaShop/issues/34925
| UI Tests          | https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/8158974105
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/34925
| Related PRs       | -
| Sponsor company   | -

### CODE REVIEW

**Comment by kpodemski** [comment]
Wrong branch.

**Review by boherm — DISMISSED** [review]

**Comment by Hlavtox** [comment]
I would keep the recommendation in the help text however, so people don't think thay can put 512 everywhere.

**Comment by M0rgan01** [comment]
Why put 160 in the help text, and 512 in the counter? Even on PPV1 the help text does not indicate the correct limit

**Comment by Hlavtox** [comment]
@M0rgan01 The fields were reduced because the new smaller limits is what meta descriptions should be, to be good for SEO.

So by rising the limit, we allow people to save older products just fine, but we allow them to make longer MEH meta things. So I would like for them to know what is the OPTIMAL length for those fields. :-)

So, this PR is perfectly OK, maxximum length for meta description is 512 chars, but optimal lengths is 160.

**Comment by M0rgan01** [comment]
@Hlavtox Ok I understand thank you. I think the help text should bring out the OPTIMAL aspect then

**Review by Hlavtox — COMMENTED** [review]
@M0rgan01 Last things I would recommend:

- There are both help and label_help_box for some reason?
- Can we also add help to meta title?
- We should improve the wording a bit, there is no guarantee that the description will show up in the search engine. I would suggest this, but I would leave it to @Julievrz (also ping @kpodemski and @MatShir)

`This description is a summary for robots, that could appear in search engines. It's best to keep it around 160 characters. If you leave it blank, an excerpt from short description will be used.`

For title:

`Public title that will appear in the browser and in search engines. It's best to keep it around 60 characters. If you leave it blank, product name will be used.`

**Comment by MatShir** [comment]
> There are both help and label_help_box for some reason?
Agree, we can remove the tool tip. 

For the content, IMO the current wording is sufficient. I'll let @Julievrz settle it.

**Comment by Hlavtox** [comment]
@MatShir Well the description is misleading a bit. `This description will appear in search engines.` is not true anymore, google doesn't show the meta description anymore in most cases, it chooses an excerpt by it's own rules.

**Comment by kpodemski** [comment]
I agree with @Hlavtox. Search engines decide what to show in the search results. Having meta tags is only a suggestion.

File: src/PrestaShopBundle/Form/Admin/Sell/Product/SEO/SEOType.php [hunk 1]
```diff
@@ -139,11 +139,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ])
             ->add('meta_description', TranslatableType::class, [
                 'label' => $this->trans('Meta description', 'Admin.Catalog.Feature'),
-                'label_help_box' => $this->trans('This description will appear in search engines. You need a single sentence, shorter than 160 characters (including spaces)', 'Admin.Catalog.Help'),
+                'label_help_box' => $this->trans('This description will appear in search engines. For optimal referencing you need a single sentence, shorter than 160 characters (including spaces)', 'Admin.Catalog.Help'),
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Product/SEO/SEOType.php]:
```suggestion
                'label_help_box' => $this->trans('For better SEO, write a sentence of 160 characters maximum (spaces included).', 'Admin.Catalog.Help'),
```

File: src/PrestaShopBundle/Form/Admin/Sell/Product/SEO/SEOType.php [hunk 2]
```diff
@@ -139,11 +139,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
                 'required' => false,
                 'type' => TextWithLengthCounterType::class,
                 'help' => $this->trans(
-                    'This description will appear in search engines. It should be a single sentence, shorter than 160 characters (including spaces).',
+                    'This description will appear in search engines. For optimal referencing you need a single sentence, shorter than 160 characters (including spaces).',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/Product/SEO/SEOType.php]:
```suggestion
                    'This description will appear in search engines. For better SEO, write a sentence of 160 characters maximum (spaces included).',
```

File: translations/default/AdminCatalogHelp.xlf [hunk 3]
```diff
@@ -358,8 +358,8 @@
   <file original="src/PrestaShopBundle/Form/Admin/Product/ProductSeo.php" source-language="en-US" target-language="en" datatype="plaintext">
     <body>
       <trans-unit id="8533d8d42f2c7ef3078797984ea14675">
-        <source>This description will appear in search engines. You need a single sentence, shorter than 160 characters (including spaces)</source>
-        <target>This description will appear in search engines. You need a single sentence, shorter than 160 characters (including spaces)</target>
+        <source>This description will appear in search engines. For optimal referencing you need a single sentence, shorter than 160 characters (including spaces)</source>
```
Comment (Julievrz) [inline translations/default/AdminCatalogHelp.xlf]:
```suggestion
        <source>This description will appear in search engines. For better SEO, write a sentence of 160 characters maximum (spaces included).</source>
```

File: translations/default/AdminCatalogHelp.xlf [hunk 4]
```diff
@@ -358,8 +358,8 @@
...
+        <target>This description will appear in search engines. For optimal referencing you need a single sentence, shorter than 160 characters (including spaces)</target>
```
Comment (Julievrz) [inline translations/default/AdminCatalogHelp.xlf]:
```suggestion
        <target>This description will appear in search engines. For better SEO, write a sentence of 160 characters maximum (spaces included).</target>
```

File: translations/default/AdminCatalogHelp.xlf [hunk 5]
```diff
@@ -552,8 +552,8 @@
         <note>Line: 117</note>
       </trans-unit>
       <trans-unit id="390348fb4642f64089e6e9ee71f25a7c">
-        <source>This description will appear in search engines. It should be a single sentence, shorter than 160 characters (including spaces).</source>
-        <target>This description will appear in search engines. It should be a single sentence, shorter than 160 characters (including spaces).</target>
+        <source>This description will appear in search engines. For optimal referencing you need a single sentence, shorter than 160 characters (including spaces).</source>
```
Comment (Julievrz) [inline translations/default/AdminCatalogHelp.xlf]:
```suggestion
        <source>This description will appear in search engines. For better SEO, write a sentence of 160 characters maximum (spaces included).</source>
```

File: translations/default/AdminCatalogHelp.xlf [hunk 6]
```diff
@@ -552,8 +552,8 @@
...
+        <target>This description will appear in search engines. For optimal referencing you need a single sentence, shorter than 160 characters (including spaces).</target>
```
Comment (Julievrz) [inline translations/default/AdminCatalogHelp.xlf]:
```suggestion
        <target>This description will appear in search engines. For better SEO, write a sentence of 160 characters maximum (spaces included).</target>
```

**Comment by Julievrz** [comment]
We can use @Hlavtox's suggestion. I've just made a few recommendations: 

`Summary for robots, that could appear in search engines. The recommended size is 160 characters maximum (including spaces). If you leave it blank, an excerpt from the short description will be used.`

For title:

`Public title that will appear in the browser and search engines. The recommended size is 160 characters maximum (including spaces). If you leave it blank, the product name will be used.`

File: translations/default/AdminCatalogHelp.xlf (hunk 3 above)
Comment (kpodemski) [reply translations/default/AdminCatalogHelp.xlf]: @Julievrz 

> >This description will appear in search engines

will -> may

what do you think? see our conversation about how search engines use meta tags

**Comment by Hlavtox** [comment]
@Julievrz Julie, you proposed two different wordings, what should be applied? 😁

File: translations/default/AdminCatalogHelp.xlf (hunk 3 above)
Comment (Julievrz) [reply translations/default/AdminCatalogHelp.xlf]: Yes, I saw the comments afterward. "may" is fine

**Review by Hlavtox — APPROVED** [review]

**Review by Hlavtox — CHANGES_REQUESTED** [review]
`will` => `may` and its good 👍

**Comment by Hlavtox** [comment]
@Julievrz Julie can we also change 
`The recommended size is 160 characters maximum (including spaces)`
to
`The recommended length is about 160 characters (including spaces)`
?

If you check the google recommendations for meta things, its just a recommended length and they show “good” examples that vary between 110-190 chars. This wording would better suit it. :-)

**Comment by ChillCode** [comment]
Well, Google will show what is in description meta, but only if it's unique and interesting, but that is a work for merchants.

Some good readings:

https://developers.google.com/search/docs/crawling-indexing/special-tags

https://developers.google.com/search/docs/appearance/snippet

https://developers.google.com/search/docs/crawling-indexing/robots-meta-tag#max-snippet

**Comment by Julievrz** [comment]
Yes that's fine by me 👍

**Comment by Hlavtox** [comment]
@M0rgan01 Morgan can you apply the maximmum > about wording to all wordings? ;)

**Review by Hlavtox — APPROVED** [review]

**Comment by Hlavtox** [comment]
Nice solution for the issue, good job @M0rgan01! :-)

**Review by boherm — APPROVED** [review]

**Comment by paulnoelcholot** [comment]
Hello @M0rgan01,

Some tests fail on auto tests, can you check?

Thanks!

**Comment by M0rgan01** [comment]
@paulnoelcholot after a restart it's good

**Review by florine2623 — APPROVED** [review]
Hello @M0rgan01 ,

Tested with more/less characters. Works as expected.

Screenshot 2024-03-15 at 16 58 43

LGTM, it is QA ✅ 
Thanks!

**Comment by mflasquin** [comment]
Thanks @M0rgan01

**Pull request merged**